etest_test/build_test.py

```diff
@@ -1,12 +1,236 @@
 """Tests for etest.build."""
 
+import logging
+import textwrap
+
 import click.testing
+import docker
+import pytest
 
 import etest.build as sut
+from etest import profile
+
+logger = logging.getLogger(__name__)
 
 
 def test_help() -> None:
     """Ensure etest-build --help exits successfully."""
     runner = click.testing.CliRunner()
     result = runner.invoke(sut.main, ["--help"])
+
+    assert result.exit_code == 0
+
+
+def test_nobuild() -> None:
+    """Ensure etest-build outputs 4 info lines when not building."""
+    runner = click.testing.CliRunner()
+    result = runner.invoke(sut.main, ["--nobuild", "-v", "INFO"])
+
+    assert result.exit_code == 0
+
+    out = [o for o in result.output.splitlines() if o]
+
+    assert len(out) == 4
+
+
+def test_invalid_push(capfd: pytest.CaptureFixture[str]) -> None:
+    """Ensure _push_image() fails when provided with an invalid profile."""
+    profile_ = profile.Profile(logger, False, "amd64", "glibc", False, True, False)
+    profile_.profile = "asdjadnancalca"  # Garbage in
+
+    push_log = sut._push_image(profile=profile_).splitlines()
+    out = [o for o in capfd.readouterr()[1].splitlines() if o]
+
+    assert len(push_log) == 2
+    assert "error" in push_log[1]
+
+    assert len(out) == 1
+
+
+@pytest.mark.slow
+def test_amd64_build() -> None:
+    """Ensure a base amd64 image builds correctly."""
+    runner = click.testing.CliRunner()
+    client = docker.APIClient(base_url="unix://var/run/docker.sock")
+    result = runner.invoke(sut.main, ["--arch", "amd64", "-v", "DEBUG"])
+    output = result.output.lower()
+
+    print(output)
+
+    assert result.exit_code == 0
+    assert "error" not in output
+
+    assert "current profile: amd64." in output
+
+    assert "building stage1 image." in output
+    assert "debug: stage1 logs:" in output
+
+    assert "building stage2 container." in output
+    assert "debug: stage2 logs:" in output
+
+    assert "committing the final image." in output
+
+    assert "cleaning up stage1 image." in output
+    assert "cleaning up stage2 container." in output
+
+    assert "build finished." in output
+
+    image_data = client.inspect_image("ebuildtest/etest:amd64")
+
+    assert image_data["Architecture"] == "amd64"
+    assert image_data["Os"] == "linux"
+
+    assert image_data["ContainerConfig"]["Cmd"] == [
+        "/bin/bash",
+        "-c",
+        textwrap.dedent(sut._libc_commands["glibc"].value)[23:-2],
+    ]
+
+    assert image_data["ContainerConfig"]["Image"] == "etest/stage1:amd64"
```

Comment:
This is a lot of assertions for the image built as expected.  This seems to check the whole implementation rather than the desired behaviour.  By checking the implementation any change to it will require a change to the test here.  That indicates a badly written test that will be ignored over time.